### DISCUSSION TITLE
Error when running V4.1.2 with HYMAP or HYMAP2

### DISCUSSION THREAD

**Discussion by Benminoungou** [NASA-LIS/LISF#1082] [General]
Hi dear all,

I'm trying to run VIC with HYMAP. In lis.config, after configuring Routing model to "HYMAP router", below is the error I got:

****************Error****************************
init routine for HYMAP router is not defined
program will seg fault.....
****************Error****************************

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

When I change  "HYMAP router" to "HYMAP2 router", the program runs until I got the error below:

****************Error****************************
set sws routine for LSM + routing instance VIC.4.1.2+HYMAP2 router is not defined
program will seg fault.....
****************Error****************************

Program received signal SIGSEGV: Segmentation fault - invalid memory reference.

I would like to know if the errors mean that HYMAP and HYMAP2 are not available for VIC.4.1.2.

Thank you in advance for your support.

/Bernard

**Comment by dmocko** [comment]
Hi @Benminoungou 

HYMAP2 is not available for VIC-4.1.2.  However, the HYMAP router should work with it, and I'm not sure why you get that error.

Can you please attach the lis.config for your "HYMAP router" simulation?

David

**Comment by bmcandr** [comment]
@dmocko, HYMAP was disabled in the default.cfg in #966. I believe HYMAP is being deprecated in favor of HYMAP2.

**Reply by dmocko** [reply]
@bmcandr - Thanks for the info.  I was looking at the public-7.3 branch, where HYMAP is still enabled.  You are correct that in the master branch it is disabled, which may be the branch that @Benminoungou is trying to use here.  It's also true that HYMAP is being deprecated in favor of HYMAP2 - while not all LSMs have yet been enabled to use HYMAP2.

**Reply by Benminoungou** [reply]
Dear @dmocko and @bmcandr, 
Thank you for these clarifications. 
Could you please tell me which models work with HYMAP2? I tried with Noah 3.3 but it doesn't work either.
Also, could I get an idea when HYMAP (or HYMAP2) will be operational with VIC4.1.2?
Best regards,
/Bernard

**Reply by agetiran** [reply]
Hi @Benminoungou,
Could you please share the lis.config for both run attempts (i.e., using Noah3.3 and VIC4.1.2)?

**Reply by Benminoungou** [reply]
Dear @agetiran,
Thank you for your reply.
Please find attached the lis.config files.
Best regards,
/Bernard
[lis.config_WA_noah_bernard.txt](https://github.com/NASA-LIS/LISF/files/8620803/lis.config_WA_noah_bernard.txt)
[lis.config_WA_VIC_bernard.txt](https://github.com/NASA-LIS/LISF/files/8620804/lis.config_WA_VIC_bernard.txt)

**Reply by dmocko** [reply]
Hi @Benminoungou 

To find which LSMs work with HYMAP2, please see this file and search for lines with "external" also containing "_hymap2":

https://github.com/NASA-LIS/LISF/blob/master/lis/plugins/LIS_lsmrouting_pluginMod.F90

LSMs that should work with HYMAP2 are: Noah-3.6, Noah-MP-3.6, Noah-MP-4.0.1, and CLSM-F2.5.

There are no plans to enable HYMAP2 in VIC-4.1.2.  We invite you to work on the code to enable this, following examples of the other LSMs.

HYMAP is being deprecated, and no further development with it will take place.

**Reply by Benminoungou** [reply]
Dear @dmocko,

Thank you for your message. I will try to work on it.

Best regards,

/Bernard
